pkg/unpack/unpacker.go

```diff
@@ -367,28 +367,28 @@ func (u *Unpacker) unpack(
 
 		select {
 		case <-ctx.Done():
-			abort()
+			abort(context.Background()) // Cleanup context
```

Comment:
I am wonder that the ctx won't work if the client doesn't set default namespace.
Maybe we can extract namespace from orignal contxt and set it into cleanup context?

Reply:
After this change I was playing around with #7861

Reply:
I like the #7861 's change. It looks great!